pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientCnx.java

```diff
@@ -508,8 +533,14 @@ protected boolean isHandshakeCompleted() {
             if (log.isDebugEnabled()) {
                 log.debug("{} Failed to add lookup-request into pending queue", requestId);
             }
-            future.completeExceptionally(new PulsarClientException.TooManyRequestsException(
-                    "Failed due to too many pending lookup requests"));
+            if (waitingLookupRequestSemaphore.tryAcquire()) {
+                waitingLookupRequests.add(Pair.of(requestId, Pair.of(request, future)));
+            } else {
+                future.completeExceptionally(new PulsarClientException.TooManyRequestsException(
```

Comment:
I think the logging here can be improved.

it is essentially saying - "There are {x} lookup request outstanding and {y} request pending", x limited by `concurrentLookupRequest` while y limited by `maxLookupRequest`